docs/design/proposals/userID-mapping.md

```diff
@@ -0,0 +1,76 @@
+# UserID Mapping
+
+In Kubernetes environments using Ceph CSI, when volumes are mounted on nodes,
+each node uses a specific userID to access the volume. Currently, there is no
+mechanism to track and expose this mapping information between nodes and userIDs
+_refer [Ceph CSI architecture](https://github.com/ceph/ceph-csi/blob/devel/docs/design/proposals/design.md)_.
+
+## Problem
+
+Without a mechanism to track userID mappings, it becomes challenging to:
+
+* Monitor which userIDs are actively being used across the cluster
+* Make informed decisions about key rotation
+* Clean up old userIDs safely
+
+## Important Note
+
+> ⚠️ **WARNING**: This feature is only effective after upgrading to a CephCSI
+> version that supports this functionality AND rebooting all nodes in the cluster.
+> With rebooting all the node, for volumes that do not have this metadata,
+> it could mean either:
+> * The volume is not currently mounted
+> * The volume was mounted with an older CephCSI version that doesn't support
+> this feature
+
+## Proposed Solution
+
+The solution involves tracking userID information by storing it in the metadata
+of RBD images and CephFS subvolumes using the key:
+`.[rbd|cephfs].csi.ceph.com/userID/<NodeId>:<userID>`
+
+## Implementation Details
+
+### NodeStageVolume()
+
+* Set the metadata on the volume:
+
+```
+.[rbd|cephfs].csi.ceph.com/userID/<NodeId>:<userID>
+```
+
+### NodeUnstageVolume()
+
+* Remove the metadata from the volume:
+
+```
+.[rbd|cephfs].csi.ceph.com/userID/<NodeId>:<userID>
```

Comment:
we will not have secrets to remove it from the NodeUnstage, please specify if you are planning to use the new secret from the configmap.

Reply:
Yes, forgot, let's do only controllerunpublish.
Linked to the secrets handling section in other design doc.